### PR TITLE
feat(check-update): add new command check-update for steamcmd game servers

### LINKED ISSUE #2001: Alert if Update available?
Is there a way that LGSM Alert me if a Serverupdate is available?
I need it for the Rustserver. If a Gameserver Update is available i cant update it instantly because there isnt instant a update available for oxide. Because of that i need to be informed if a update is available.

Thanks for you answer.

### PR DESCRIPTION
**Pull request by h3o66**
# Description

Adding a command to alert for an update for the server if it is updated by the steamcmd.
I did also a refactor to move all steamcmd related functions to a new core functions

Fixes #2001 

## Type of change

* [ ] Bug fix (a change which fixes an issue).
* [x] New feature (change which adds functionality).
* [ ] New Server (new server added).
* [x] Refactor (restructures existing code).
* [ ] Comment update (typo, spelling, explanation, examples, etc).

## Checklist

PR will not be merged until all steps are complete.

* [x] This pull request links to an issue.
* [x] This pull request uses the `develop` branch as its base.
* [x] This pull request Subject follows the Conventional Commits standard.
* [x] This code follows the style guidelines of this project.
* [x] I have performed a self-review of my code.
* [x] I have checked that this code is commented where required.
* [x] I have provided a detailed with enough description of this PR.
* [ ] I have checked If documentation needs updating.

## Documentation

If documentation does need updating either update it by creating a PR (preferred) or request a documentation update.
* User docs: https://github.com/GameServerManagers/LinuxGSM-Docs
* Dev docs: https://github.com/GameServerManagers/LinuxGSM-Dev-Docs

**Thank you for your Pull Request!**

### CODE REVIEW

**Review by dgibbs64 — APPROVED** [review]
Checked this and happy with the changes including creating core_steamcmd

**Comment by dgibbs64** [comment]
Have the docs been updated to add the new command?

**Comment by h3o66** [comment]
> Have the docs been updated to add the new command?

Added PR for the docs:
https://github.com/GameServerManagers/LinuxGSM-Docs/pull/53

File: lgsm/functions/core_steamcmd.sh [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+functionselfname="$(basename "$(readlink -f "${BASH_SOURCE[0]}")")"
+
+fn_install_steamcmd(){
+	if [ "${shortname}" == "ark" ]&&[ "${installsteamcmd}" == "1" ]; then
+		steamcmddir="${serverfiles}/Engine/Binaries/ThirdParty/SteamCMD/Linux"
+	fi
+	if [ ! -d "${steamcmddir}" ]; then
+		mkdir -p "${steamcmddir}"
+	fi
+	remote_fileurl="${1}"
```
Comment (h3o66) [inline lgsm/functions/core_steamcmd.sh:16]: @dgibbs64 do you know if these are needed at all here ?
There is a open issue on shellcheck side.

Comment (dgibbs64) [reply lgsm/functions/core_steamcmd.sh:16]: I can't see any shellcheck issues relating to this if statement

**Review by dgibbs64 — APPROVED** [review]
Happy with this

**Pull request merged**